jkube-kit/jkube-kit-wildfly-jar/src/test/java/org/eclipse/jkube/wildfly/jar/generator/WildflyJARGeneratorTest.java

```diff
@@ -27,154 +27,163 @@
 
-    @TempDir
-    Path temporaryFolder;
-
-    @Mocked
     private GeneratorContext context;
 
-    @Mocked
     private JavaProject project;
-
+    @BeforeEach
+    public void setUp() throws Exception {
+        context = mock(GeneratorContext.class);
+        project = mock(JavaProject.class);
+    }
     @Test
-    void notApplicable() {
+    void notApplicable() throws IOException {
         WildflyJARGenerator generator = new WildflyJARGenerator(createGeneratorContext());
         assertThat(generator.isApplicable(Collections.emptyList())).isFalse();
     }
 
     // To be revisited if we enable jolokia and prometheus.
     @Test
-    void getEnv() {
+    void getEnv() throws IOException {
         WildflyJARGenerator generator = new WildflyJARGenerator(createGeneratorContext());
         Map<String, String> extraEnv = generator.getEnv(true);
         assertThat(extraEnv).isNotNull().hasSize(4);
     }
     
     @Test
-    void getExtraOptions() {
+    void getExtraOptions() throws IOException {
         WildflyJARGenerator generator = new WildflyJARGenerator(createGeneratorContext());
         List<String> extraOptions = generator.getExtraJavaOptions();
         assertThat(extraOptions).isNotNull()
             .singleElement().isEqualTo("-Djava.net.preferIPv4Stack=true");
     }
     
     @Test
-    void slimServer(@Mocked final JavaProject project) throws IOException {
-      Map<String, Object> options = new HashMap<>();
-      Map<String, String> pluginOptions = new HashMap<>();
-      options.put(PLUGIN_OPTIONS, pluginOptions);
-      pluginOptions.put(JBOSS_MAVEN_DIST, null);
-      pluginOptions.put(JBOSS_MAVEN_REPO, "target" + File.separator + "myrepo");
-
-      Path targetDir = Files.createDirectory(temporaryFolder.resolve("target"));
-      Files.createDirectory(targetDir.resolve("myrepo"));
-
-      GeneratorContext ctx = contextForSlimServer(project, options, temporaryFolder);
-      WildflyJARGenerator generator = new WildflyJARGenerator(ctx);
-
-      List<String> extraOptions = generator.getExtraJavaOptions();
-      assertThat(extraOptions).isNotNull()
-          .hasSize(2)
-          .satisfies(option -> assertThat(option).first()
-              .isEqualTo("-Djava.net.preferIPv4Stack=true"))
-          .satisfies(option -> assertThat(option).last()
-              .isEqualTo("-Dmaven.repo.local=/deployments/myrepo"));
+    void slimServer() throws IOException {
+        Map<String, Object> options = new HashMap<>();
+        Map<String, String> pluginOptions = new HashMap<>();
+        options.put(PLUGIN_OPTIONS, pluginOptions);
+        pluginOptions.put(JBOSS_MAVEN_DIST, null);
+        pluginOptions.put(JBOSS_MAVEN_REPO, "target" + File.separator + "myrepo");
+        //
+        Path tmpDir = Files.createTempDirectory("bootable-jar-test-project");
+        Path targetDir = tmpDir.resolve("target");
+        Path repoDir = targetDir.resolve("myrepo");
+        Files.createDirectories(repoDir);
```

Comment:
This was OK as it was, also, the finally clause deleting these files is not necessary. The extensions takes care of this.